### PR TITLE
Feat: Login, Signup screens and authentication

### PR DESCRIPTION
**Pull request by vanshitalavange**
Overview:
-  screens for login and signup 
- authentication for login and signup 

Preview:

https://user-images.githubusercontent.com/86543154/160684016-f0b21f89-4e63-4461-babf-06762a8de44e.mp4

### CODE REVIEW

File: src/pages/auth/login.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const submitLoginForm = (event) => {
+    event.preventDefault();
+    loginHandler(loginCredentials);
+  };
+  return (
+    <section className="login flex-col-center">
+      <h2>Sign in</h2>
+      <form onSubmit={submitLoginForm} className="login-form flex-column">
+        <div className="input-field-wrapper flex-column">
+          <label for="email">Email</label>
```
Comment (sadathshariff) [inline src/pages/auth/login.jsx]: React Ignores For Attribute instead use htmlFor

**Review by sadathshariff — APPROVED** [review]
Overall Looks Good

File: src/components/ResponsiveNavbar/ResponsiveNavbar.jsx [hunk 2]
```diff
@@ -66,23 +71,34 @@ export function ResponsiveNavbarForTablet() {
...
+          class="fa fa-user header-icon header-badge-icon align-end"
+        ></i>
+      </li>
+      <button
+        onClick={() => {
+          localStorage.removeItem("token");
+          setIsLoggedIn(false);
+        }}
+        className="btn-logout display-none"
+        style={{ display: isLoggedIn ? "block" : "none" }}
```
Comment (nishasen) [inline src/components/ResponsiveNavbar/ResponsiveNavbar.jsx:98]: I like how you are toggling between styles

File: src/components/header/header.css [hunk 3]
```diff
@@ -82,6 +82,15 @@
   cursor: pointer;
   color: var(--primary-color);
 }
+.btn-logout{
+  color:var(--secondary-color);
```
Comment (nishasen) [inline src/components/header/header.css:86]: good use of variables here

File: src/pages/auth/login.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    try {
+      const response = await axios.post("/api/auth/login", data);
+      if (response.status === 200) {
+        setIsLoggedIn(true);
+        setUserDetails(response.data.foundUser);
+        localStorage.setItem("token", response.data.encodedToken);
+        navigate("/");
+      }
+    } catch (error) {
+      setErrorMsg("YOUR EMAIL IS NOT REGISTERED WITH US😔");
```
Comment (nishasen) [inline src/pages/auth/login.jsx:35]: you can give a alert box here, if the user wants to signup

File: src/pages/auth/login.jsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            id="email"
+            className="form-input"
+            name="email"
+            type="email"
+            placeholder="user@example.com"
+            required
+          />
+        </div>
+        <div className="input-field-wrapper flex-column">
+          <label for="password">Password</label>
```
Comment (nishasen) [inline src/pages/auth/login.jsx]: instead of for use htmlFor

**Review by nishasen — APPROVED** [review]
I liked your approaches how you are checking if the user is logged in or not, good and clean code, have suggested some small things, have a look into them

File: src/pages/auth/login.jsx (hunk 5 above)
Comment (vanshitalavange) [reply src/pages/auth/login.jsx]: yup changes done

File: src/pages/auth/login.jsx (hunk 1 above)
Comment (vanshitalavange) [reply src/pages/auth/login.jsx]: yup changes done

File: src/pages/auth/login.jsx (hunk 4 above)
Comment (vanshitalavange) [reply src/pages/auth/login.jsx:35]: I'll keep this for now as I have the sign up link just below it, but definitely once done with everything I'll implement this

**Pull request merged**